frontend/src/utilities/useEnableApplication.tsx

```diff
@@ -59,66 +65,115 @@ export const useEnableApplication = (
-                : EnableApplicationStatus.FAILED,
-              error: response.valid ? '' : response.error,
+        if (isInternalRouteIntegrationsApp(internalRoute)) {
+          getIntegrationAppEnablementStatus(internalRoute)
+            .then((response) => {
+              if (!response.isInstalled && response.canInstall) {
+                watchHandle = setTimeout(watchStatus, 10 * 1000);
+                return;
+              }
+              if (response.isInstalled && !response.canInstall) {
+                setEnableStatus({
+                  status: EnableApplicationStatus.SUCCESS,
+                  error: '',
+                });
+                dispatchResults(undefined);
+              }
+            })
+            .catch((e) => {
+              if (!cancelled) {
+                setEnableStatus({ status: EnableApplicationStatus.FAILED, error: e.message });
+              }
+              dispatchResults(e.message);
             });
-            dispatchResults(response.valid ? undefined : response.error);
-          })
-          .catch((e) => {
-            if (!cancelled) {
-              setEnableStatus({ status: EnableApplicationStatus.FAILED, error: e.message });
-            }
-            dispatchResults(e.message);
-          });
+        } else {
+          getValidationStatus(appId)
+            .then((response) => {
+              if (!response.complete) {
+                watchHandle = setTimeout(watchStatus, 10 * 1000);
+                return;
+              }
+              setEnableStatus({
+                status: response.valid
+                  ? EnableApplicationStatus.SUCCESS
+                  : EnableApplicationStatus.FAILED,
+                error: response.valid ? '' : response.error,
+              });
+              dispatchResults(response.valid ? undefined : response.error);
+            })
+            .catch((e) => {
+              if (!cancelled) {
+                setEnableStatus({ status: EnableApplicationStatus.FAILED, error: e.message });
+              }
+              dispatchResults(e.message);
+            });
+        }
       };
       watchStatus();
     }
     return () => {
       cancelled = true;
       clearTimeout(watchHandle);
     };
-  }, [appId, dispatchResults, enableStatus.status]);
+  }, [appId, dispatchResults, enableStatus.status, internalRoute]);
 
   React.useEffect(() => {
     let closed = false;
     if (doEnable) {
-      postValidateIsv(appId, enableValues)
-        .then((response) => {
-          if (!closed) {
-            if (!response.complete) {
-              setEnableStatus({ status: EnableApplicationStatus.INPROGRESS, error: '' });
-              return;
+      if (internalRoute && isInternalRouteIntegrationsApp(internalRoute)) {
+        enableIntegrationApp(internalRoute, enableValues)
+          .then((response) => {
+            if (!closed) {
+              if (!response.isAppEnabled && response.canEnable) {
```

Comment:
When you change the backend implementation for status -- we should clean this up too.

```suggestion
              if (!response.isInstalled && response.canEnable) {
```

Reply:
updated